src/action_unexpanded.ml

```diff
@@ -301,11 +338,11 @@ module Infer = struct
       | Run (prog, _) -> acc +<! prog
       | Redirect (_, fn, t)  -> infer (acc +@ fn) t
       | Cat fn               -> acc +< fn
-      | Write_file (fn, _)  -> acc +@ fn
-      | Rename (src, dst)    -> acc +< src +@ dst
+      | Write_file (fn, _)  -> acc +@+ fn
+      | Rename (src, dst) -> acc +<+ src +@+ dst
```

Comment:
Isn't this kinda broken though? 
`src` is clearly more than just a dependency, as you pointed out.
It's probably an intermediate file (so neither a target nor a dependency). 
I'm surprised this works.

Reply:
We have some code that handles paths that are both targets and dependencies. The workaround is to treat such paths as just targets.